app/views/root/local_transaction.html.erb

```diff
@@ -58,3 +58,15 @@
     </div>
   </section>
 <% end %>
+
+<% if @postcode_error %>
+  <script type="text/javascript">
+      GOVUK.analytics.trackEvent(
+        "userAlerts:LocalTransaction",
+        "postcodeErrorShown:<%= @postcode_error %>",
+        {
+          label: "<%= t @location_error || 'Please enter a valid full UK postcode.' %>"
+        }
+      )
+  </script>
```

Comment:
My preference would be a way of sending analytics data through `data-` attributes. There is a pattern for this in the `govuk_admin_template` that I'm currently looking at porting to the frontend toolkit.